release/train_tests/benchmark/image_classification/image_classification_jpeg/factory.py

```diff
@@ -0,0 +1,238 @@
+            region=AWS_REGION,
+            connect_timeout=connection_timeout,
+            request_timeout=request_timeout,
+        )
+        return s3fs
+
+    def get_ray_datasets(self) -> Dict[str, ray.data.Dataset]:
+        """Get Ray datasets for training and validation.
+
+        Creates training and validation datasets with:
+        1. Partitioning by class for efficient data loading
+        2. Image preprocessing with optional random transforms
+        3. Resource allocation for concurrent validation
+        4. Row limits based on benchmark configuration
+
+        Returns:
+            Dictionary containing:
+                - "train": Training dataset with random transforms
+                - "val": Validation dataset without transforms
+        """
+        # Configure S3 filesystem connection
+        s3fs = self.get_s3fs_with_boto_creds()
+
+        # Create training dataset with class-based partitioning
+        train_pattern = IMAGENET_JPEG_SPLIT_S3_DIRS["train"]
+        train_partitioning = Partitioning(
+            "dir", base_dir=train_pattern, field_names=["class"]
+        )
+        train_ds = (
+            ray.data.read_images(
+                train_pattern,
+                mode="RGB",
+                include_paths=True,
+                partitioning=train_partitioning,
+                filesystem=s3fs,
+            )
+            .limit(self.benchmark_config.limit_training_rows)
+            .map(get_preprocess_map_fn(random_transforms=True))
+        )
+
+        # Create validation dataset with same partitioning
+        val_pattern = IMAGENET_JPEG_SPLIT_S3_DIRS["train"]
+        val_partitioning = Partitioning(
+            "dir", base_dir=val_pattern, field_names=["class"]
+        )
+        val_ds = (
+            ray.data.read_images(
+                val_pattern,
+                mode="RGB",
+                include_paths=True,
+                partitioning=val_partitioning,
+                filesystem=s3fs,
+            )
+            .limit(self.benchmark_config.limit_validation_rows)
+            .map(get_preprocess_map_fn(random_transforms=False))
+        )
+
+        # Configure resource allocation for concurrent validation
+        if self.benchmark_config.validate_every_n_steps > 0:
+            cpus_to_exclude = 16
+            train_ds.context.execution_options.exclude_resources = (
+                train_ds.context.execution_options.exclude_resources.add(
+                    ray.data.ExecutionResources(cpu=cpus_to_exclude)
+                )
+            )
+            val_ds.context.execution_options.resource_limits = (
+                ray.data.ExecutionResources(cpu=cpus_to_exclude)
+            )
+            log_with_context(
+                f"Reserving {cpus_to_exclude} CPUs for validation that happens "
+                "concurrently with training every "
+                f"{self.benchmark_config.validate_every_n_steps} steps"
+            )
```

Comment:
I think we can delete this stuff for now. We don't do concurrent validation since I wasn't able to get it running in a performant way.